### PR TITLE
Mock response payload generation improved to allow users to specify required response.

### PR DESCRIPTION
**Pull request by VinulaUthsara**
Fixes : https://github.com/wso2/product-apim/issues/7783

Mock Response Payload Generation feature only provides a response related to a single response code (lowest response code). 

This has been improved to support the response of multiple response code. Allowing users to specify which response code they would like to get the payload from. By supplying : ?responseCode=<response code> in cURL users can specify desired response code.

Screenshot from 2020-05-26 20-56-29

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 1]
```diff
@@ -246,8 +244,50 @@ private String getGeneratedResponseVar(String responseCode, String example, Stri
...
+
+                " if (responseCode == null) {\n\n"+
+                "  mc.setProperty('CONTENT_TYPE', 'application/"+minResponseType+"');\n"+
+                "  mc.setPayload"+minResponseType.toUpperCase()+"(response"+minResponseCode+minResponseType+");\n\n"+
+                "} else "+
+
+                "{\n\n"+
+                "  mc.setProperty('CONTENT_TYPE', 'application/json');\n" +
+                "  mc.setPayloadJSON(response501json);\n\n"+
+                "}";
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:290]: Format the code properly.

Comment (VinulaUthsara) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java [hunk 2]
```diff
@@ -291,8 +285,49 @@ private String getGeneratedResponseVar(String responseCode, String example, Stri
...
+                "\n\"description\" : " + "\"Not Implemented\"\n" +
+                "}\n\n" +
+                "var responseCode = mc.getProperty('query.param.responseCode');\n\n" +
+                responseSectionString +
+                " if (responseCode == null) {\n\n" +
+                "  mc.setProperty('HTTP_SC', \"" + minResponseCode + "\");\n" +
+                "  mc.setProperty('CONTENT_TYPE', 'application/" + minResponseType + "');\n" +
+                "  mc.setPayload" + minResponseType.toUpperCase() + "(response" + minResponseCode + minResponseType + ");\n\n" +
+                "} else " +
+                "{\n\n" +
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:327]:
```suggestion
                "} else {\n\n" +
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 3]
```diff
@@ -156,21 +157,19 @@ private void setOtherSchemes(List<String> otherSchemes) {
                         if (applicationJson != null) {
                             String jsonExample = Json.pretty(applicationJson);
                             genCode.append(getGeneratedResponseVar(responseEntry, jsonExample, "json"));
-                            if (responseCode == minResponseCode && !setPayloadResponse){
-                                responseSection.append(getGeneratedSetResponse(responseEntry, "json"));
-                                setPayloadResponse = true;
-                                if (applicationXml != null) {
-                                    responseSection.append("\n\n/*").append(getGeneratedSetResponse(responseEntry, "xml")).append("*/\n\n");
-                                }
+                            responseSection.append(getGeneratedIFsforCodes(responseEntry, getGeneratedSetResponse(responseEntry, "json")));
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:160]:
```suggestion
                            responseSection.append(getGeneratedConditionsForResponseCodes(responseEntry, getGeneratedSetResponse(responseEntry, "json")));
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 4]
```diff
@@ -156,21 +157,19 @@ private void setOtherSchemes(List<String> otherSchemes) {
...
+                            if (responseCode == minResponseCode) {
+                                minResponseType = ("json");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:162]: Do we need these brackets?
```suggestion
                                minResponseType = "json";
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 5]
```diff
@@ -246,8 +244,49 @@ private String getGeneratedResponseVar(String responseCode, String example, Stri
...
+                "\n\"description\" : " + "\"Not Implemented\"\n" +
+                "}\n\n" +
+                "var responseCode = mc.getProperty('query.param.responseCode');\n\n" +
+                responseSectionString +
+                " if (responseCode == null) {\n\n" +
+                "  mc.setProperty('HTTP_SC', \"" + minResponseCode + "\");\n" +
+                "  mc.setProperty('CONTENT_TYPE', 'application/" + minResponseType + "');\n" +
+                "  mc.setPayload" + minResponseType.toUpperCase() + "(response" + minResponseCode + minResponseType + ");\n\n" +
+                "} else " +
+                "{\n\n" +
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:286]:
```suggestion
                "} else {\n\n" +
```

**Pull request merged**